File: services/ink-engine/app/models/incident.py
"""告警记录模型"""
import uuid
from datetime import datetime
from sqlmodel import SQLModel, Field


class Incident(SQLModel, table=True):
    __tablename__ = "incidents"

    id: str = Field(default_factory=lambda: str(uuid.uuid4()), primary_key=True)
    monitor_id: str = Field(foreign_key="monitors.id")
    level: str = Field(default="P1")
    summary: str
    ai_diagnosis: str | None = None
    first_action: str | None = None
    opened_at: datetime = Field(default_factory=datetime.utcnow)
    recovered_at: datetime | None = None

File: apps/desktop/src-tauri/src/models/incident.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Incident {
    pub id: String,
    pub monitor_id: String,
    pub level: String,
    pub summary: String,
    pub ai_diagnosis: Option<String>,
    pub first_action: Option<String>,
    pub opened_at: Option<String>,
    pub recovered_at: Option<String>,
}

impl Incident {
    pub fn from_row(row: &rusqlite::Row) -> rusqlite::Result<Self> {
        Ok(Self {
            id: row.get("id")?,
            monitor_id: row.get("monitor_id")?,
            level: row.get("level")?,
            summary: row.get("summary")?,
            ai_diagnosis: row.get("ai_diagnosis")?,
            first_action: row.get("first_action")?,
            opened_at: row.get("opened_at")?,
            recovered_at: row.get("recovered_at")?,
        })
    }
}
